[GStreamer][MSE] Enable ManagedMediaSource

vivienne-w opened the pull request:
#### 7e1e9cbe78acfcba005e9488a9335db667909876
<pre>
[GStreamer][MSE] Enable ManagedMediaSource
<a href="https://bugs.webkit.org/show_bug.cgi?id=280052">https://bugs.webkit.org/show_bug.cgi?id=280052</a>

Reviewed by Philippe Normand.

Managed MSE should be purely a multiplatform thing, so enabling it is straightforward.

* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultManagedMediaSourceEnabled): Enable Managed MSE.

Canonical link: <a href="https://commits.webkit.org/292812@main">https://commits.webkit.org/292812@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/234927b633c41124a7bc14f56d9bcb0abf4bcdf0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97060 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16683 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6891 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102141 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47585 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16975 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25133 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73942 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31154 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100063 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12806 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87787 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54278 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12557 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46915 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82628 "Found 10 new API test failures: TestWebKitAPI.ProcessSwap.NavigatingToLockdownMode, TestWebKitAPI.LockdownMode.NotAllowedFont, TestWebKitAPI.ServiceWorkers.LockdownModeInServiceWorkerProcess, TestWebKitAPI.LockdownMode.SVGFonts, TestWebKitAPI.LockdownMode.AllowedFont, TestWebKitAPI.ProcessSwap.LockdownModeSystemSettingChange, TestWebKitAPI.LockdownMode.NotSupportedFontLoadingAPI, TestWebKitAPI.LockdownMode.AllowedFontLoadingAPI, TestWebKitAPI.ProcessSwap.CannotDisableLockdownModeWithoutBrowserEntitlement, TestWebKitAPI.LockdownMode.NotAllowedFontLoadingAPI (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5721 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/104162 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24134 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/17608 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82990 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/24385 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83894 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82386 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20747 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27075 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4623 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/17639 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24099 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/29249 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23922 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27234 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25495 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WTF/wtf/PlatformEnableGlib.h:
```diff
@@ -65,4 +65,5 @@
 
 #if ENABLE(WPE_PLATFORM) || PLATFORM(GTK)
 #define ENABLE_DAMAGE_TRACKING 1
+#define ENABLE_MANAGED_MEDIA_SOURCE 1
```
Don't we need an ifdef in `defaultManagedMediaSourceEnabled()` ?

ntrrgc replied:
Where is this `#define` read?

philn replied:
ENABLE(FOO) expands to ENABLE_FOO

philn reviewed (approved):
Indeed I forgot that the ManagedMSE ENABLE ifdef was removed.

vivienne-w committed:
[GStreamer][MSE] Enable ManagedMediaSource
https://bugs.webkit.org/show_bug.cgi?id=280052

Reviewed by Philippe Normand.

Managed MSE should be purely a multiplatform thing, so enabling it is straightforward.

* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultManagedMediaSourceEnabled): Enable Managed MSE.

Canonical link: https://commits.webkit.org/292812@main
